massa-factory-worker/src/denunciation_factory.rs

```diff
@@ -0,0 +1,260 @@
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::thread;
+
+use crossbeam_channel::{select, Receiver};
+use tracing::{debug, info, warn};
+
+use massa_factory_exports::{FactoryChannels, FactoryConfig};
+use massa_models::block_header::SecuredHeader;
+use massa_models::denunciation::Denunciation;
+use massa_models::endorsement::SecureShareEndorsement;
+use massa_models::slot::Slot;
+use massa_models::timeslots::get_closest_slot_to_timestamp;
+use massa_time::MassaTime;
+
+// TODO: rework these values
+const DENUNCIATION_FACTORY_ENDORSEMENT_CACHE_MAX_LEN: usize = 4096;
+const DENUNCIATION_FACTORY_BLOCK_HEADER_CACHE_MAX_LEN: usize = 4096;
+
+/// Structure gathering all elements needed by the factory thread
+#[allow(dead_code)]
+pub(crate) struct DenunciationFactoryWorker {
+    cfg: FactoryConfig,
+    channels: FactoryChannels,
+    factory_receiver: Receiver<()>,
```

Comment:
The type is not very meaningful to me.
Particularly for this one, I'd like a comment.

Reply:
added